xdsl/dialects/builtin.py

```diff
@@ -45,8 +45,7 @@ def verify(self, attr: Attribute) -> None:
 class ArrayAttr(GenericData[tuple[AttributeCovT, ...]]):
     name: str = "array"
 
-    def __init__(self: ArrayAttr[AttributeCovT],
```

Comment:
not sure why the self was annotated here, `Pylance` suddenly started screaming after the change in ir.py, and this change calmed it down...